### PR TITLE
Optimize tsm1.FloatDecoder

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

This commit changes the `FloatDecoder.val` from a `float64` type to a `uint64` to avoid an additional type conversion during read. Now the type gets converted to a `float64` only on call to `Values()`.

/cc @jwilder 

### Benchmark

```
benchmark                                   old ns/op     new ns/op     delta
BenchmarkEngine_CreateIterator_Count_1M     12145578      11839383      -2.52%

benchmark                                   old allocs     new allocs     delta
BenchmarkEngine_CreateIterator_Count_1M     74             73             -1.35%
```

###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by jwilder** [comment]
```
[ERROR] run: Command 'go test -v -parallel 1 -timeout 480s ./...' failed with error: # github.com/influxdata/influxdb/tsdb/engine/tsm1
tsdb/engine/tsm1/engine_test.go:22:2: cannot find package "github.com/pkg/profile" in any of:
	/usr/local/go/src/github.com/pkg/profile (from $GOROOT)
	/root/go/src/github.com/pkg/profile (from $GOPATH)
```

File: tsdb/engine/tsm1/engine_test.go [hunk 1]
```diff
@@ -541,6 +542,7 @@ func benchmarkEngineCreateIteratorLimit(b *testing.B, pointN int) {
 
 func benchmarkIterator(b *testing.B, opt influxql.IteratorOptions, pointN int) {
 	e := MustInitBenchmarkEngine(pointN)
+	defer profile.Start(profile.ProfilePath("/tmp"), profile.CPUProfile).Stop()
```
Comment (mark-rushakoff) [inline tsdb/engine/tsm1/engine_test.go]: Did you mean to check in this line?

Comment (benbjohnson) [inline tsdb/engine/tsm1/engine_test.go]: No, I keep forgetting to remove that. :)

Comment (benbjohnson) [inline tsdb/engine/tsm1/engine_test.go]: Fixed: e6a57f2

**Comment by benbjohnson** [comment]
@jwilder I removed the `pkg/profile` in e6a57f2.

**Comment by benbjohnson** [comment]
@jwilder Can I merge this?

**Comment by jwilder** [comment]
👍

**Pull request merged**